File: 0461-hamming-distance/python/Solution.py
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        r = x ^ y
        mask = 1
        count = 0
        for i in range(31):
            if (r & mask) != 0:
                count += 1
            mask *= 2
        return count

File: 0461-hamming-distance/cpp/Solution.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
        int r = x ^ y;
        long mask = 1;
        int count = 0;
        for (int i = 0; i < 31; ++i) {
            if ((r & mask) != 0) {
                count++;
            }
            mask *= 2;
        }
        return count;
    }
};
